app/configurator/components/chart-controls/create-color-palette.tsx

```diff
@@ -0,0 +1,191 @@
+import { useEvent } from "@dnd-kit/utilities";
+import { t } from "@lingui/macro";
+import { Trans } from "@lingui/react";
+import { Box, Button, IconButton, Typography } from "@mui/material";
+import { makeStyles } from "@mui/styles";
+import { nanoid } from "nanoid";
+import { forwardRef, useState } from "react";
+
+import Flex from "@/components/flex";
+import { Input } from "@/components/form";
+import { CustomPaletteType } from "@/config-types";
+import SvgIcClose from "@/icons/components/IcClose";
+import ColorPaletteCreator from "@/login/components/color-palettes/color-palette-types";
+import { ColorItem, defaultColorValues } from "@/palettes";
+import { theme } from "@/themes/federal";
+import { createCustomColorPalette } from "@/utils/chart-config/api";
+import { useMutate } from "@/utils/use-fetch-data";
+
+import { DRAWER_WIDTH } from "../drawer";
+
+const useStyles = makeStyles({
+  autocompleteMenuContent: {
+    "--mx": "1rem",
+    "--colorBoxSize": "1.25rem",
+    "--columnGap": "0.75rem",
+    width: DRAWER_WIDTH,
+  },
+  autocompleteHeader: {
+    margin: "1rem var(--mx)",
+  },
+  autocompleteApplyButtonContainer: {
+    padding: "1rem var(--mx)",
+    background: "rgba(255,255,255,0.75)",
+  },
+  textInput: {
+    margin: `${theme.spacing(4)} 0px`,
+    padding: "30px",
+    width: "100%",
+    height: 40,
+    minHeight: 40,
+  },
+  autocompleteApplyButton: {
+    justifyContent: "center",
+  },
+});
+
+export const DrawerContent = forwardRef<
+  HTMLDivElement,
+  {
+    onClose: (palette?: CustomPaletteType) => void;
+    type: CustomPaletteType["type"];
+    customColorPalettes?: CustomPaletteType[];
+  }
+>((props, ref) => {
+  const { onClose, type, customColorPalettes } = props;
```

Comment:
Nit: I think it would be good to rename it to something like `ColorPaletteDrawerContent`, as it's not anymore in the same file where it was originally used. Maybe also file name could better indicate that we are exporting color palette drawer content? 🤔